test/units/TEST-03-JOBS.sh

```diff
@@ -166,4 +168,30 @@ assert_rc 3 systemctl --quiet is-active succeeds-on-restart.target
 systemctl start fails-on-restart.target || :
 assert_rc 3 systemctl --quiet is-active fails-on-restart.target
 
+# Test shortcutting auto restart
+
+export UNIT_NAME="TEST-03-JOBS-shortcut-restart.service"
+TMP_FILE="/tmp/test-03-shortcut-restart-test$RANDOM"
+
+cat >"/run/systemd/system/$UNIT_NAME" <<EOF
+[Service]
+Type=oneshot
+ExecStart=rm -v "$TMP_FILE"
+Restart=on-failure
+RestartSec=1d
+RemainAfterExit=yes
```

Comment:
Missing daemon-reload?

Reply:
There's no need for daemon-reload when a completely new unit is installed? IOW they'd be gc'd immediately anyway?

Reply:
Oh, really? I did not know that. So, can we drop many daemon-reload in our integration tests??